ksql-parser/src/main/java/io/confluent/ksql/parser/rewrite/StatementRewriter.java

```diff
@@ -102,508 +92,245 @@ protected Node visitArithmeticBinary(
+        node.getLocation(),
+        (Expression) process(node.getBase(), context),
+        node.getFieldName()
+    );
   }
 
-  protected Node visitNullLiteral(final NullLiteral node, final Object context) {
-    // use an if/else block here (instead of isPresent.map(...).orElse(...)) so only one object
-    // gets instantiated (issue #1784)
-    if (node.getLocation().isPresent()) {
-      return new NullLiteral(node.getLocation().get());
-    } else {
-      return new NullLiteral();
-    }
+  protected Node visitLiteral(final Literal node, final Object context) {
+    return node;
   }
 
   protected Node visitArithmeticUnary(final ArithmeticUnaryExpression node, final Object context) {
-    // use an if/else block here (instead of isPresent.map(...).orElse(...)) so only one object
-    // gets instantiated (issue #1784)
-    if (node.getLocation().isPresent()) {
-      return new ArithmeticUnaryExpression(
-          node.getLocation().get(),
-          node.getSign(),
-          (Expression) process(node.getValue(), context)
-      );
-    } else {
-      return new ArithmeticUnaryExpression(
-          node.getSign(),
-          (Expression) process(node.getValue(), context)
-      );
-    }
+    final Expression rewrittenExpression = (Expression) process(node.getValue(), context);
+
+    return new ArithmeticUnaryExpression(
+        node.getLocation(),
+        node.getSign(),
+        rewrittenExpression
+    );
   }
 
   protected Node visitNotExpression(final NotExpression node, final Object context) {
-    // use an if/else block here (instead of isPresent.map(...).orElse(...)) so only one object
-    // gets instantiated (issue #1784)
-    if (node.getLocation().isPresent()) {
-      return new NotExpression(node.getLocation().get(),
-          (Expression) process(node.getValue(), context)
-      );
-    } else {
-      return new NotExpression((Expression) process(node.getValue(), context));
-    }
+    return new NotExpression(
+        node.getLocation(),
+        (Expression) process(node.getValue(), context)
+    );
   }
 
   protected Node visitSingleColumn(final SingleColumn node, final Object context) {
     return node.copyWithExpression((Expression) process(node.getExpression(), context));
   }
 
   protected Node visitAllColumns(final AllColumns node, final Object context) {
-    // use an if/else block here (instead of isPresent.map(...).orElse(...)) so only one object
-    // gets instantiated (issue #1784)
-    if (node.getLocation().isPresent()) {
-      if (node.getPrefix().isPresent()) {
-        return new AllColumns(node.getLocation().get(), node.getPrefix().get());
-      } else {
-        return new AllColumns(node.getLocation().get());
-      }
-
-    } else {
-      if (node.getPrefix().isPresent()) {
-        return new AllColumns(node.getLocation().get());
-      } else {
-        throw new KsqlException("Cannot have both location and prefix null in AllColumns AST "
```

Comment:
this check disappeared. is it necessary?

Reply:
No,  not necessary.  The `AllCoumns` is being created with the correct fields and is immutable, so no need to copy.

The location is always kind of optional anyway. Really, we should be providing it in production code to provide better error messages, but its not used otherwise.  The prefix is also optional. So this is actually a bit of a weird check.